libtenzir/include/tenzir/plugin.hpp

```diff
@@ -145,7 +145,10 @@ class component_plugin : public virtual plugin {
 
   /// Components that should be created before the current one so initialization
   /// can succeed.
-  /// Defaults to empty list.
+  /// Note that the *only* guarantee made is that components are able to
+  /// retrieve actor handles of the wanted components from the registry, but
+  /// there is no guarantee that the initialization routine is already
+  /// completed. Defaults to empty list.
```

Comment:
I don't understand this comment. I assume that with "initialization routine" you're referring to the code up to returning the behavior. But that runs in its own thread, and any message sent to the actor handle will only arrive after the initial behavior was returned. So what is this warning the developer about?

Reply:
If there is a request happening before the behavior is returned, there is no guarantee that the request is already in the receiving actors inbox. Or in other words, if there are two components who depend on each other there's no guarantee that such requests from the one that is first in the list are arriving before the second.

I'll try to reword the comment to that effect.